### PR TITLE
replaced serde_cbor with minicbor-ser library

### PR DESCRIPTION
**Pull request by integral-llc**
### Motivation

serde_cbor is not maintained anymore, so we replaced it with another library compatible with serde.

### In this PR
* < Additions, removals, fixes, features >

https://github.com/mobilecoinfoundation/mobilecoin/issues/1260

### CODE REVIEW

File: util/serial/Cargo.toml [hunk 1]
```diff
@@ -9,9 +9,9 @@ edition = "2018"
 # because rmp-ser../../alloc doesn't build against ser../../std,
 # so if anything else in your build plan will activate ser../../std, then mcseri../../std is
 # required.
-std = [ "serde/std", "serde_cbor/std" ]
+std = [ "serde/std" ]
```
Comment (remoun) [inline util/serial/Cargo.toml:12]: IIUC we want to pull in that package's `std` feature
```suggestion
std = [ "serde/std", "minicbor-ser/std" ]
```

File: util/serial/src/lib.rs [hunk 2]
```diff
@@ -55,15 +56,15 @@ pub fn serialize<T: ?Sized>(value: &T) -> Result<Vec<u8>, encode::Error>
 where
     T: serde::ser::Serialize + Sized,
 {
-    Ok(serde_cbor::to_vec(value)?)
+    Ok(cbor::to_vec(value)?)
 }
 
 // Forward mc_util_serial::deserialize to bincode::deserialize
```
Comment (remoun) [inline util/serial/src/lib.rs:62]:
```suggestion
```

File: util/serial/src/lib.rs [hunk 3]
```diff
@@ -55,15 +56,15 @@ pub fn serialize<T: ?Sized>(value: &T) -> Result<Vec<u8>, encode::Error>
 where
     T: serde::ser::Serialize + Sized,
 {
-    Ok(serde_cbor::to_vec(value)?)
+    Ok(cbor::to_vec(value)?)
```
Comment (remoun) [inline util/serial/src/lib.rs:59]: This method's comment comment is stale, would you mind updating it while we're here?

**Review by remoun — APPROVED** [review]

**Comment by garbageslam** [comment]
How did we decide between minicbor and ciborium?

https://docs.rs/minicbor/latest/minicbor/index.html
https://docs.rs/minicbor/latest/minicbor/index.html

**Comment by integral-llc** [comment]
tried `minicbor_ser` and `ciborium`, none of them worked for our test cases. Abandoning this task till we decide how to add Protobuff

**Comment by eranrund** [comment]
Out of curiosity, what failed?

**Comment by integral-llc** [comment]
mainly all tests related to `serde_` serialization, i.e. Report tests.

**Pull request closed without merging**